Dockerfile.windows

```diff
@@ -1,57 +1,96 @@
+#
+# 'sleep 5' is a deliberate workaround for a current problem on containers in Windows 
+# Server 2016. It ensures that the network is up and available for when the command is
+# network related. This bug is being tracked internally at Microsoft and exists in TP4.
+# Generally sleep 1 or 2 is probably enough, but making it 5 to make the build file
+# as bullet proof as possible. This isn't a big deal as this only runs the first time.
+#
+# The cygwin posix utilities from GIT aren't usable interactively as at January 2015. This
+# is because they require a console window which isn't present in a container in Windows.
+# See the example at the top of this file. Do NOT use -it in that docker run!!! 
+#
+# Don't try to use a volume for passing the source through. The cygwin posix utilities will
+# balk at reparse points. Again, see the example at the top of this file on how use a volume
+# to get the built binary out of the container.
 
 FROM windowsservercore
 
-ENV GO_VERSION=1.5.3 \
+# Environment variable notes:
+#  - GOLANG_VER should be updated to be consistent with the Linux dockerfile.
+#  - FROM_DOCKERFILE is used for detection of building within a container.
+ENV GOLANG_VERSION=1.5.3 \
     GIT_VERSION=2.7.0 \
-    RSRC_COMMIT=e48dbf1b7fc464a9e85fcec450dddf80816b76e0 \
+    RSRC_COMMIT=ba14da1f827188454a4591717fff29999010887f \
     GOPATH=C:/go;C:/go/src/github.com/docker/docker/vendor \
     FROM_DOCKERFILE=1
 
+# Make sure we're in temp for the downloads
 WORKDIR c:/windows/temp
 
-RUN powershell -command \
-    sleep 2 ; \
-    iex ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1'))
-
-RUN powershell -command \
-    sleep 2 ; \
-    choco install curl -y -v ; \
-    choco install git -y -v --version=%GIT_VERSION% -params '"/GitAndUnixToolsOnPath"'
-
-RUN setx /M Path "c:\ProgramData\chocolatey\bin;c:\Program Files\Git\usr\bin;%Path%;c:\gcc\bin"
+# Chocolatey and install curl/git
+RUN powershell -command sleep 5 ; iex ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1'))
+RUN powershell -command sleep 5; choco install curl -y -v
+RUN powershell -command sleep 5; choco install git -y -v --version=%GIT_VERSION% -params '"/GitAndUnixToolsOnPath"'
 
-RUN powershell -command \
-    sleep 2; \
-    curl.exe -L -k -o go.msi https://storage.googleapis.com/golang/go%GO_VERSION%.windows-amd64.msi; \
-    curl.exe -L -k -o gcc.zip http://downloads.sourceforge.net/project/tdm-gcc/TDM-GCC%205%20series/5.1.0-tdm64-1/gcc-5.1.0-tdm64-1-core.zip ; \
-    curl.exe -L -k -o runtime.zip http://downloads.sourceforge.net/project/tdm-gcc/MinGW-w64%20runtime/GCC%205%20series/mingw64runtime-v4-git20150618-gcc5-tdm64-1.zip ; \
-    curl.exe -L -k -o binutils.zip http://downloads.sourceforge.net/project/tdm-gcc/GNU%20binutils/binutils-2.25-tdm64-1.zip
+# Path
+RUN setx /M Path "c:\ProgramData\chocolatey\bin;c:\Program Files\Git\usr\bin;%Path%;c:\gcc\bin;c:\7zip"
```

Comment:
I was trying to keep layers to minimum. But no, at least I need one path at this point. It made more sense putting all the paths into one line.